reloader.py

```diff
@@ -132,103 +189,289 @@ def add_action(
-
+        
         if whats_this is not None:
             action.setWhatsThis(whats_this)
-
+        
         if add_to_toolbar:
             # Adds plugin icon to Plugins toolbar
             self.iface.addToolBarIcon(action)
-
+        
         if add_to_menu:
             self.iface.addPluginToMenu(self.menu, action)
-
+        
         self.actions.append(action)
-
+        
         return action
+    # END def addAction(...):
 
     def initGui(self):
         """Create the menu entries and toolbar icons inside the QGIS GUI."""
-
+        
         icon_path = ":/plugins/reloader/layer-reload.png"
-        self.add_action(
+        self.addAction(
             icon_path,
             text=self.tr("Reload selected layer(s)"),
-            callback=self.reload,
+            callback=self.reloadCallback,
             parent=self.iface.mainWindow(),
         )
-
+        
         icon_path = ":/plugins/reloader/layer-reopen.png"
-        self.add_action(
+        self.addAction(
             icon_path,
             text=self.tr("Reopen selected layer(s)"),
-            callback=self.reopen,
+            callback=self.reopenCallback,
             parent=self.iface.mainWindow(),
         )
-
+        
         icon_path = ":/plugins/reloader/layer-watch.png"
-        self.add_action(
+        self.addAction(
             icon_path,
             text=self.tr("Start watching layer(s) for changes"),
-            callback=self.watch,
+            callback=self.watchCallback,
             parent=self.iface.mainWindow(),
         )
-
+        
         icon_path = ":/plugins/reloader/layer-unwatch.png"
-        self.add_action(
+        self.addAction(
             icon_path,
             text=self.tr("Stop watching layer(s) for changes"),
-            callback=self.unwatch,
+            callback=self.unwatchCallback,
             parent=self.iface.mainWindow(),
         )
+        
+        # Create status indicator for watched layers
+        # All such layers share a single indicator object
+        self.indicator = QgsLayerTreeViewIndicator()
+        cur_path = os.path.dirname(os.path.abspath(__file__))
+        icon_path = os.path.join(cur_path, "layer-watched-large.png")
+        icon = QIcon(icon_path)
+        self.indicator.setIcon(icon)
+        
+        # Connect to the project load event
+        # When a project is loaded this will reconnect all of the file watches
+        # that were save in the project file.
+        self.iface.projectRead.connect(self.reconnectWatches)
+        
+        # Connect to layer tree layer added event
+        # This is called whenever one or more layers is added to the project.
+        # Our code will automatically add watches to added layers that are
+        # backed by local files that are already being watched.
+        project = QgsProject.instance()
+        if project is not None:
+            project.legendLayersAdded.connect( self.getLegendLayersAddedCallback() )
```

Comment:
Why not just https://qgis.org/pyqgis/3.40/core/QgsProject.html#qgis.core.QgsProject.layersAdded?